[Agent Loop] Signal-based cancel for agent message workflow

philipperolet opened the pull request:
## Description
Fixes https://github.com/dust-tt/tasks/issues/4273 and https://github.com/dust-tt/tasks/issues/3814

- Update `cancelMessageGenerationEvent` to signal the associated agent loop workflow by ID, removing existing behavior (Redis cancel key and DB status update to "cancelled").
- Change `makeAgentLoopWorkflowId` to take `workspaceId`, `conversationId`, `agentMessageId` as args.
- Add `cancel_agent_loop_signal` and handle it in `agentLoopWorkflow` by cancelling in-flight activities using a `CancellationScope` and finishing. Run a non-cancellable finalization activity on cancellation that handles what was previously done.

## Risks
Blast radius: Agent conversation cancel path (front + agent Temporal worker), workflow ID callers, cancellation event publishing.
Risk: standard

## Deploy Plan
- Deploy front.
- No database migrations required.

aubin-tchoi commented on front/lib/api/assistant/pubsub.ts:
```diff
@@ -101,37 +106,49 @@ export async function cancelMessageGenerationEvent(
           workspaceId: auth.getNonNullableWorkspace().id,
           sId: messageId,
         },
+        include: [
+          {
+            model: ConversationModel,
+            as: "conversation",
+            required: true,
+            attributes: ["sId"],
+          },
+        ],
```
why not pass the `conversationId` in the arguments? when calling from the front-end we most likely have it

aubin-tchoi commented on front/temporal/agent_loop/lib/workflow_ids.ts:
```diff
@@ -2,12 +2,10 @@ import type { AuthenticatorType } from "@app/lib/auth";
 export function makeAgentLoopWorkflowId(
-  authType: AuthenticatorType,
-  runAgentArgs: RunAgentAsynchronousArgs
+  workspaceId: string | null,
+  conversationId: string,
+  agentMessageId: string
```
can we turn these in a structured object? 🙏🏻

philipperolet commented:
Addressed review feedback:\n- Replaced console.error with logger.error in agent_loop worker (GEN8).\n- Re-ran Prettier, ESLint, and tsc on front; all clean.\n\nPlease take another look. 🙏

philipperolet commented:
Addressed comments from @aubin-tchoi:\n- Narrowed try/catch to just  and removed nested try/catch.\n- Added bounded concurrency via  in cancel path.\n-  now accepts  and callers pass it.\n- Renamed local vars to , , .\n- Switched  to a structured arg with non-null .\n- Finalizer: inlined one-off helper and switched to  for the cancellation log.\n- Sorted imports (lint) and re-ran Prettier, ESLint, and tsc — all clean.\n\nPlease take another look. 🙏

aubin-tchoi commented on front/temporal/agent_loop/lib/workflow_ids.ts:
```diff
@@ -1,13 +1,15 @@
+  conversationId,
+  agentMessageId,
+}: {
+  workspaceId: string; // must be non-null workspace sId
```
```suggestion
  workspaceId: string;
```

aubin-tchoi commented on front/lib/api/assistant/pubsub.ts:
```diff
@@ -99,44 +105,40 @@ export async function* getConversationEvents({
+      // We use the message id provided by the caller as the agentMessageId.
+      const agentMessageId = messageId;
+
+      if (workspaceId && conversationId && agentMessageId) {
```
I think you don't need this `if`

aubin-tchoi reviewed (approved):
LGTM ✨
